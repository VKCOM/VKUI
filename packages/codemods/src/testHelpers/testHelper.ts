/**
 * Файл для упрощенной работы с дефолтными функциями запуска `jscodeshift`
 */
import path from 'path';
import {
  applyTransform,
  defineSnapshotTestFromFixture as defineSnapshotTestFromFixtureOrig,
  // @ts-expect-error: TS7016 no types for package
} from 'jscodeshift/dist/testUtils';
import { JSCodeShiftOptions } from '../types';

export { applyTransform };

export function getTestFixturesInputPath(
  dirName: string,
  testFilePrefix: string,
  extension = 'tsx',
) {
  const fixtureDir = path.join(dirName, '..', '__testfixtures__');
  return path.join(fixtureDir, testFilePrefix + `.input.${extension}`);
}

/**
 * У оригинального defineSnapshotTestFromFixture нет типов
 * Избавляемся от необходимости в тестах постоянно указывать `module`
 */
export function defineSnapshotTestFromFixture(
  dirName: string,
  transformName: string,
  options: JSCodeShiftOptions,
  testFilePrefix: string,
) {
  // Assumes transform is one level up from __tests__ directory
  const module = require(path.join(dirName, '..', transformName));
  defineSnapshotTestFromFixtureOrig(
    dirName,
    module,
    options,
    testFilePrefix,
    'transforms correctly',
    { parser: 'tsx' },
  );
}
