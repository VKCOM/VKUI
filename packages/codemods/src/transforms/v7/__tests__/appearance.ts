jest.autoMockOff();

import { defineSnapshotTestFromFixture } from '../../../testHelpers/testHelper';

const name = 'appearance';
const fixtures = ['basic', 'alias'] as const;

describe(name, () => {
  fixtures.forEach((test) =>
    defineSnapshotTestFromFixture(__dirname, name, global.TRANSFORM_OPTIONS, `${name}/${test}`),
  );
});
