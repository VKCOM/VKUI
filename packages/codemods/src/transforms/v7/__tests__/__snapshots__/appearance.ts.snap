// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`appearance transforms correctly 1`] = `
"import { ColorScheme, type ColorSchemeType, useColorScheme } from '@vkontakte/vkui';
import React from 'react';

const Component: React.FC<{
  appearance: ColorSchemeType;
}> = ({
  appearance,
}) => {
  return (
    <div></div>
  )
}

const App = () => {
  const appearance: ColorSchemeType = ColorScheme.LIGHT;

  const fromHookAppearance = useColorScheme();

  return (
    (<React.Fragment>
      <Component appearance={ColorScheme.DARK} />
    </React.Fragment>)
  );
};"
`;

exports[`appearance transforms correctly 2`] = `
"import { ColorScheme as AppearanceAlias, type ColorSchemeType as AppearanceTypeAlias } from '@vkontakte/vkui';
import React from 'react';

const Component: React.FC<{
  appearance: AppearanceTypeAlias;
}> = () => {
  return (
    <div></div>
  )
}

const App = () => {
  const appearance: AppearanceTypeAlias = AppearanceAlias.LIGHT;

  return (
    <React.Fragment>
      <Component appearance={AppearanceAlias.DARK} />
    </React.Fragment>
  );
};"
`;
